sys/include/phydat.h

```diff
@@ -0,0 +1,175 @@
+ * @{
+ *
+ * @file
+ * @brief       Generic data container for physical data interface
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef SECT_DATA_H
+#define SECT_DATA_H
+
+#include <stdint.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   The fixed number of dimensions we work with
+ *
+ * We use a fixed number of 3 dimensions, as many physical values we encounter
+ * can be expressed this way. In practice we have e.g. readings from
+ * accelerometers, gyros, color sensors, or set data for RGB LEDs.
+ *
+ * When expressing 1-dimensional data we just ignore the 2 higher dimension.
+ * This leads to a slight overhead of some byte of memory - but we benefit from
+ * a unified data structure for passing around physical data.
+ */
+#define PHYDAT_DIM                  (3U)
+
+/**
+ * @brief   The maximum length of a scaling string
+ */
+#define PHYDAT_SCALE_STR_MAXLEN     (sizeof("*E-128\0"))
+
+/**
+ * @brief   Definition of physical units and comparable data types
+ *
+ * This list should contain all needed physical units (e.g. SI units), but also
+ * non-physical units that can be used to define the type of data passed around.
+ * This can be for example BOOL or aggregate values. As rule of thumb, the unit
+ * list can contain anything that helps two modules automatically negotiate, if
+ * they can understand each other.
+ *
+ * @note    Extent this list as needed.
+ */
+enum {
+    /* generic values */
+    UNIT_UNDEF,     /**< unit undefined */
+    UNIT_NONE,      /**< data has no physical unit */
+    /* temperature */
+    UNIT_TEMP_C,    /**< degree Celsius */
+    UNIT_TEMP_F,    /**< degree Fahrenheit */
+    UNIT_TEMP_K,    /**< Kelvin */
+    /* dimension */
+    UNIT_M,         /**< meters */
+    UNIT_M2,        /**< square meters */
+    UNIT_M3,        /**< cubic meters */
+    /* kinetic */
+    UNIT_G,         /**< gravitational force */
+    UNIT_DPS,       /**< degree per second */
+    /* weight */
+    UNIT_GR,        /**< grams - not using the SI unit (kg) here to make scale
+                     *   handling simpler */
+    /* electricity */
+    UNIT_A,         /**< Ampere */
+    UNIT_V,         /**< Volts */
+    UNIT_GS,        /**< gauss */
+    /* pressure */
+    UNIT_BAR,       /**< Beer? */
+    UNIT_PA,        /**< Pascal */
+    /* light */
+    UNIT_CD,        /**< Candela */
+    /* logical */
+    UNIT_BOOL,      /**< boolean value [0|1] */
+    UNIT_PERCENT,   /**< out of 100 */
+    UNIT_PERMILL,   /**< out of 1000 */
+    UNIT_PPM,       /**< part per million */
+    /* aggregate values */
+    UNIT_TIME,      /**< the three dimensions contain sec, min, and hours */
+    UNIT_DATE       /**< the 3 dimensions contain days, months and years */
+    /* extend this list as needed */
+};
+
+/**
+ * @brief   Generic data structure for expressing physical values
+ *
+ * Physical data is express in a 3-dimensional touple of values. In addition to
```

Comment:
`s/express/expressed/`

Sorry for being that late, I waited with reading the docs until now that SAUL is merged.